overlord/devicestate/handlers_install.go

```diff
@@ -274,11 +300,36 @@ func (m *DeviceManager) doSetupRunSystem(t *state.Task, _ *tomb.Tomb) error {
 	var installedSystem *install.InstalledSystemSideData
 	// run the create partition code
 	logger.Noticef("create and deploy partitions")
+
+	// Find out installed components in the temporary system, which will be
+	// the same as those snapd will install in the final system.
```

Comment:
is this true though?  components can have modes too in the model

Reply:
Now this uses the seed for the different install cases